Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/threshold/BasicThresholder.java

```diff
@@ -16,41 +16,39 @@
 package com.amazon.randomcutforest.parkservices.threshold;
 
 import static com.amazon.randomcutforest.CommonUtils.checkArgument;
+import static com.amazon.randomcutforest.RandomCutForest.DEFAULT_SAMPLE_SIZE;
+import static com.amazon.randomcutforest.RandomCutForest.DEFAULT_SAMPLE_SIZE_COEFFICIENT_IN_TIME_DECAY;
 import static java.lang.Math.max;
+import static java.lang.Math.min;
+import static java.lang.Math.sqrt;
 
 import java.util.List;
 
-import lombok.Getter;
-import lombok.Setter;
-
+import com.amazon.randomcutforest.config.TransformMethod;
 import com.amazon.randomcutforest.parkservices.statistics.Deviation;
+import com.amazon.randomcutforest.util.Weighted;
 
-@Getter
-@Setter
 public class BasicThresholder {
 
-    public static double DEFAULT_THRESHOLD_PERSISTENCE = 0.5;
+    public static double DEFAULT_SCORE_DIFFERENCING = 0.5;
     public static int DEFAULT_MINIMUM_SCORES = 10;
-    public static double DEFAULT_ABSOLUTE_SCORE_FRACTION = 0.5;
-    public static double DEFAULT_UPPER_THRESHOLD = 2.0;
-    public static double DEFAULT_LOWER_THRESHOLD = 1.0;
-    public static double DEFAULT_LOWER_THRESHOLD_NORMALIZED = 0.9;
-    public static double DEFAULT_LOWER_THRESHOLD_ONED = 1.1;
+    public static double DEFAULT_LOWER_THRESHOLD = 0.9;
+    public static double DEFAULT_LOWER_THRESHOLD_NORMALIZED = 0.8;
+    public static double DEFAULT_ABSOLUTE_THRESHOLD = 0.7;
     public static double DEFAULT_INITIAL_THRESHOLD = 1.5;
-    public static double DEFAULT_Z_FACTOR = 2.5;
-    public static double DEFAULT_UPPER_FACTOR = 5.0;
-    public static boolean DEFAULT_AUTO_ADJUST_LOWER_THRESHOLD = false;
-    public static double DEFAULT_THRESHOLD_STEP = 0.1;
+    public static double DEFAULT_Z_FACTOR = 3.0;
+    public static double MINIMUM_Z_FACTOR = 2.0;
+    public static boolean DEFAULT_AUTO_THRESHOLD = true;
+    public static int DEFAULT_DEVIATION_STATES = 3;
 
     // keeping a count of the values seen because both deviation variables
     // primaryDeviation
-    // and secondaryDeviation may not be used always -- in current code, they are
-    // both used always
+    // and secondaryDeviation may not be used always
     protected int count = 0;
 
     // horizon = 0 is short term, switches to secondary
     // horizon = 1 long term, switches to primary
-    protected double thresholdPersistence = DEFAULT_THRESHOLD_PERSISTENCE;
+    protected double scoreDifferencing = DEFAULT_SCORE_DIFFERENCING;
```

Comment:
Horizon is an old term - it was used as a balance point between long term standard deviation and standard deviation of score differencing. Setting horizon to 1 meant using long term standard deviation only (with a likely lower recall, but likely higher precision). But now, in presence of forecasting -- there is a clear notion of (forecasting) horizon. So the score differencing is an explicit name to disambiguate. In term of relationships, scoreDifferencing = 1 - horizon. This should be a non-issue for the default settings since the default was 0.5.